vscode-extension/src/aiConfigEditor.ts

```diff
@@ -0,0 +1,413 @@
+    // startServer.stderr.on("data", (data) => {
+    //   this.extensionOutputChannel.error(this.prependMessage(data, document));
+    //   console.error(`server stderr: ${data}`);
+    // });
+
+    startServer.on("spawn", () => {
+      this.extensionOutputChannel.info(
+        this.prependMessage(
+          `Started server at port=${openPort}, pid=${startServer.pid}`,
+          document
+        )
+      );
+      console.log(`server spawned: ${startServer.pid}`);
+    });
+
+    startServer.on("close", (code) => {
+      if (code !== 0) {
+        this.extensionOutputChannel.error(
+          this.prependMessage(
+            `Server at port=${openPort}, pid=${startServer.pid} was terminated unexpectedly with exit code ${code}`,
+            document
+          )
+        );
+        console.error(`server terminated unexpectedly: exit code=${code}`);
+      } else {
+        this.extensionOutputChannel.info(
+          this.prependMessage(
+            `Server at port=${openPort}, pid=${startServer.pid} was terminated successfully.`,
+            document
+          )
+        );
+        console.log(`server terminated successfully`);
+      }
+    });
+
+    startServer.on("error", (err) => {
+      this.extensionOutputChannel.error(
+        this.prependMessage(JSON.stringify(err), document)
+      );
+      // TODO: saqadri - add "restart" option with error message
+    });
+
+    const serverUrl = `http://localhost:${openPort}`;
+
+    return {
+      proc: startServer,
+      url: serverUrl,
+    };
+  }
+
+  /**
+   * Get the static html used for the editor webviews.
+   */
+  private getHtmlForWebview(
+    webview: vscode.Webview,
+    editorServer: ServerInfo
+  ): string {
+    // The JS file from the React build output
+    const scriptUri = getUri(
+      webview,
+      this.context.extensionUri,
+      "editor",
+      "build",
+      "static",
+      "js",
+      "main.js"
+    );
+
+    // Use a nonce to whitelist which scripts can be run
+    const nonce = getNonce();
+
+    // Tip: Install the es6-string-html VS Code extension to enable code highlighting below
+    return /*html*/ `
+	 <!DOCTYPE html>
+	 <html lang="en">
+	   <head>
+		 <meta charset="utf-8">
+		 <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">
+		 <meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src data: http: https: 'self'; connect-src vscode-webview: ${editorServer.url} http: https:; style-src 'unsafe-inline' ${webview.cspSource} https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0 https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/editor/editor.main.css; script-src 'nonce-${nonce}' vscode-resource: https: http: https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0 https://cdn.jsdelivr.net; font-src https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/base/browser/ui/codicons/codicon/codicon.ttf; worker-src blob:;">
+		 <title>Hello World</title>

```

Comment:
Does this title actually matter? If so, should it be the filename?

Reply:
Doesn't really matter, I can remove it too